recipes/sonic-cpp/all/conanfile.py

```diff
@@ -0,0 +1,77 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
+import os
+
+required_conan_version = ">=1.53.0"
+
+class SonicCppConan(ConanFile):
+    name = "sonic-cpp"
+    description = "A fast JSON serializing & deserializing library, accelerated by SIMD."
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/bytedance/sonic-cpp"
+    topics = ("json", "parser", "writer", "serializer", "deserializer", "header-only")
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "8",
+            "clang": "7",
+            "apple-clang": "12",
+        }
+
+    def requirements(self):
+        # TODO: required string-view_lite only on __cplusplus < 201703L
+        self.requires("string-view-lite/1.7.0", transitive_headers=True)
```

Comment:
Having this check for cppstd would mean that the library would always be tested with this dependency in CI, as it only builds the first valid version it finds, and in this case it would be 11 so the dependency would always be pulled, but I _think_ it's ok? In the general user case when using cpp >= 17 there just won't be any dependency pulled and if that fails then it's _just_ an upstream error

```suggestion
        if valid_min_cppstd(self, 17):
            self.requires("string-view-lite/1.7.0")
```

(No need to define transitive headers for `header-library` either, [see the relevant docs here])(https://docs.conan.io/2/reference/conanfile/methods/requirements.html#package-type-trait-inferring)